patt/src/main/java/org/example/deepcopy/People.java

```diff
@@ -0,0 +1,41 @@
+package org.example.deepcopy;
+
+import org.example.deepcopy.Address;
+
+public class People implements Cloneable {
+    private String name;
+    private Address adress;
+    public People(String name, Address adress) {
+        this.name = name;
+        this.adress = adress;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Address getAdress() {
+        return adress;
+    }
+
+    public void setAdress(Address adress) {
+        this.adress = adress;
+    }
+
+    @Override
+    public Object clone() throws CloneNotSupportedException {
+        return super.clone();
```

Comment:
In this way, you do shallow copy. To make deep copy here, you have to create a new object with new address based on state of current object.
There is a code practic in lecture meeting record.